### PR TITLE
Add pack-missing-tests script to make it easy to spot missing pack tests.

### PR DESCRIPTION
**Pull request by jjm**
### Status 

Functional - Request for comment

### Description

Adds a script that makes it easy to spot missing or extra tests for packs that have changed..

### Checklist (tick everything that applies)

- [x] Code linting (required, can be done after the PR checks)

### CODE REVIEW

**Comment by Kami** [comment]
Nice :+1: 

Can you paste an example output of this script?

**Comment by Kami** [comment]
And since it appears to be informational only at this point, we should also update Travis config so it also runs on Travis.

File: scripts/pack-missing-tests.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+if [ -n "${1}" ]; then
+    DIR=${1}
+else
+    echo "Need a pack dir"
+    exit 1
+fi
+
+ACTIONS=$(cd ${DIR}/actions;ls -1 *.py)
```
Comment (Kami) [inline scripts/pack-missing-tests.sh]: Yeah, that's not 100% percent accurate, but I'm fine with it for now.

**Comment by jjm** [comment]
I've cleaned up the output a little, here's some examples:

```bash
====== packs/alertlogic ======

Actions missing tests:
        scan_get_results_by_title.py
        scan_get_results.py
        scan_list_scan_executions.py
        scan_list_scans.py

Tests with no actions:

Coverage:
        Actions: 0/4
```

```bash
====== packs/orion/ ======

Actions missing tests:
        node_create.py
        node_discover_and_add_interfaces.py
        node_remanage.py
        node_unmanage.py

Tests with no actions:
        test_action_aliases.py

Coverage:
        Actions: 3/6
```

```bash
====== packs/duo ======

Actions missing tests:
        auth_auth.py
        auth_check.py
        auth_ping.py

Tests with no actions:

Coverage:
        Actions: 0/3
```

```bash
====== packs/cloudflare ======

Actions missing tests:

Tests with no actions:

Coverage:
        Actions: 1/1
```

File: scripts/pack-missing-tests.sh (hunk 1 above)
Comment (jjm) [inline scripts/pack-missing-tests.sh]: I've changed that to PACKS, which is hopefully more correct. I had both PACK and DIR at one point. However that did not work when it came to adding the Make target.

**Comment by jjm** [comment]
The new output format can be extended to cover sensors too, if the file name of alias tests follows the manner as actions and sensors it can report on that too :-D

**Comment by jjm** [comment]
I'm hoping this will be useful for others, it started out a quick hack to spot test coverage holes in the Orion pack PR I'm currently working now.

**Comment by Kami** [comment]
@jjm Great, thanks.

Eventually, as @tonybaloney has pointed out in the past, it would be great if we can move all those scripts in some common place in st2 repository so users outside st2contrib can utilize them as well, but for now I'm fine with those scripts only living in this repository.

**Comment by Kami** [comment]
I'll merge this into master and hook it up to Travis inside master.

Thanks again.

**Comment by jjm** [comment]
No worries, automatic reporting of missing tests will be rather nice.

Some way of supporting scripts for testing / supporting packs outside of st2contrib would be very nice. I need to set up CI on our internal pack and a standard way of doing it would be great.

**Comment by Kami** [comment]
Made some improvements to the script (c5214c10b7638c7b2902609ac7fe1889337f7832) and enabled Travis check (f33d761d8bb76be7c88ee64ff48162343710ff08).

**Comment by tonybaloney** [comment]
@Kami fwiw, SaltStack has done a reasonable job at this by creating a separate package on pypi for testing.

**Comment by Kami** [comment]
@tonybaloney Yeah, we created st2sdk in the past (https://github.com/stackstorm/st2sdk) which we planned to release as a separate package on PyPi.

We could potentially add those scripts and tools there or create a separate package (I can't really think of a good name for it from top of my head).

**Pull request merged**